tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -128,19 +136,45 @@ public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
                 if (commandType == CommandType)
                 {
                     // use the cached values if command.GetType() == typeof(TCommand)
-                    return DbScopeFactory.CreateDbCommandScope(tracer, command, IntegrationId, DbTypeName, OperationName);
+                    return DbScopeFactory.CreateDbCommandScope(tracer, command, IntegrationId, DbTypeName, OperationName, GetServiceName(tracer, DbTypeName));
                 }
 
                 // if command.GetType() != typeof(TCommand), we are probably instrumenting a method
                 // defined in a base class like DbCommand and we can't use the cached values
                 if (TryGetIntegrationDetails(commandType.FullName, out var integrationId, out var dbTypeName))
                 {
                     var operationName = $"{dbTypeName}.query";
-                    return DbScopeFactory.CreateDbCommandScope(tracer, command, integrationId.Value, dbTypeName, operationName);
+                    return DbScopeFactory.CreateDbCommandScope(tracer, command, integrationId.Value, dbTypeName, operationName, GetServiceName(tracer, dbTypeName));
                 }
 
                 return null;
             }
+
+            private static string GetServiceName(Tracer tracer, string dbTypeName)
+            {
+                if (!tracer.Settings.TryGetServiceName(dbTypeName, out string serviceName))
+                {
+                    if (DbTypeName != dbTypeName)
+                    {
+                        // We cannot cache in the base class
+                        return $"{_tracerDefaultServiceName}-{dbTypeName}";
+                    }
+
+                    // If not a base class
+                    if (_tracerDefaultServiceName == tracer.DefaultServiceName)
+                    {
+                        // Service has not changed
+                        return _serviceName;
+                    }
+
+                    // We create or replace the cache with the new service name
+                    _tracerDefaultServiceName = tracer.DefaultServiceName;
+                    serviceName = $"{_tracerDefaultServiceName}-{dbTypeName}";
+                    _serviceName = serviceName;
```

Comment:
tiniest nit: should we set `_tracerDefaultServiceName ` at the end so we don't use the _old_ service name in case of 2 simultaneous requests?
```suggestion
                    serviceName = $"{_tracerDefaultServiceName}-{dbTypeName}";
                    _serviceName = serviceName;
                    _tracerDefaultServiceName = tracer.DefaultServiceName;
```

Reply:
done in 0036942